src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -138,16 +143,29 @@ impl<'a, C: Client<T>, T: Timestamp + Lattice> StorageController<'a, C, T> {
             }))
             .await
     }
-    pub async fn allow_source_compaction(&mut self, frontiers: Vec<(GlobalId, Antichain<T>)>) {
+    pub async fn allow_source_compaction(
+        &mut self,
+        frontiers: Vec<(GlobalId, Antichain<T>)>,
+    ) -> Result<(), StorageError> {
+        // Validate that the ids exist.
+        self.validate_ids(frontiers.iter().map(|(id, _)| *id))?;
```

Comment:
NIT: `.iter().cloned()` (as the other cases) reads much better

Reply:
It needs to drop the `Antichain<T>` that is the other argument.